build.sh

```diff
@@ -14,4 +14,4 @@ cmake \
   -S src \
   -B build
 cmake build -L
-cmake --build build -- -j$(($(nproc) - 1))
+cmake --build build -v -j$(($(nproc) - 1))
```

Comment:
You can remove the -v now if you want, was useful for debugging. You don't want the "--".

Reply:
Thanks for all of this @henryiii. The addition of `--` was my doing (sorry). Can you explain a bit though? 

```console
cmake --build
Usage: cmake --build [<dir> | --preset <preset>] [options] [-- [native-options]]
Options:
  <dir>          = Project binary directory to be built.
  --preset <preset>, --preset=<preset>
                 = Specify a build preset.
  --list-presets
                 = List available build presets.
  --parallel [<jobs>], -j [<jobs>]
                 = Build in parallel using the given number of jobs. 
                   If <jobs> is omitted the native build tool's 
                   default number is used.
                   The CMAKE_BUILD_PARALLEL_LEVEL environment variable
                   specifies a default parallel level when this option
                   is not given.
  --target <tgt>..., -t <tgt>... 
                 = Build <tgt> instead of default targets.
  --config <cfg> = For multi-configuration tools, choose <cfg>.
  --clean-first  = Build target 'clean' first, then build.
                   (To clean only, use --target 'clean'.)
  --verbose, -v  = Enable verbose output - if supported - including
                   the build commands to be executed. 
  --             = Pass remaining options to the native tool.
```

so I assumed that passing the reamaining options with `--` would be fine. You'd just recommend 

```cmake
cmake --build build -j$(($(nproc) - 1))
```

or

```cmake
cmake --build build --clean-first -j$(($(nproc) - 1))
```
?

Reply:
@matthewfeickert the `-j` is listed by `--parallel` so it's an option handled by `cmake` especially, separate from the other native tool options. Technically `-j` is also supported by the native tool as well.

Reply:
Yes, it's better to have CMake handle this than using passthrough. That way it works with all build tools and is simpler. You could use CMAKE_PARALLEL_BUILD_LEVEL instead, if you wanted a variable.